## 7.3 跳转语句

### 7.3.1 break语句

遇到break就会跳出循环体。  

例如1：  

当增加break语句时：  

    for item in range(10):
        if item == 3:
            break
        print('Count is {0}'.format(item))

    Count is 0
    Count is 1
    Count is 2

例如2：  

当没有break语句时：  

    for item in range(10):
    # if item == 3:
    #     break
    print('Count is {0}'.format(item))

    Count is 0
    Count is 1
    Count is 2
    Count is 3
    Count is 4
    Count is 5
    Count is 6
    Count is 7
    Count is 8
    Count is 9

### 7.3.2 continue语句

遇到continue语句结束当次循环，continue之后的语句不再执行，接着执行下次循环。  

例如1：  

    for item in range(10):
        if item == 3:
            continue
        print('Count is {0}'.format(item))

    Count is 0
    Count is 1
    Count is 2
    Count is 4
    Count is 5
    Count is 6
    Count is 7
    Count is 8
    Count is 9

从运行结果中可以看到，只有3的情况没有执行到。

### 7.3.3 while和for中的else语句

在while和for中的else语句的目的是当循环正常结束时执行else之后的语句，当在遇到break、return和异常抛出时，属于非正常结束，则不执行else之后的语句。  

例如1：  

    i = 0

    while i * i < 10:
        i += 1
        print('{0} X {0} = {1}'.format(i, i, i * i))
    else:
        print('While Over!')

    print('------------------------')

    1 X 1 = 1
    2 X 2 = 2
    3 X 3 = 3
    4 X 4 = 4
    While Over!
    ------------------------

这是一个正常的循环体加else语句。

例如2：

    while i * i < 10:
        i += 1
        if i == 3:
            break
        print('{0} X {0} = {1}'.format(i, i, i * i))
    else:
        print('While Over!')

    print('------------------------')

    1 X 1 = 1
    2 X 2 = 2
    ------------------------

从运行结果可以看出，当i为3时，程序跳出了正常的循环，而且并没有执行else语句组中的内容。

例如3：

    while i * i < 10:
        i += 1
        if i == 3:
            continue
        print('{0} X {0} = {1}'.format(i, i, i * i))
    else:
        print('While Over!')

    print('------------------------')


    1 X 1 = 1
    2 X 2 = 2
    4 X 4 = 4
    While Over!
    ------------------------

从程序运行结果可以看出，当i为3时，程序结束了当此的循环而继续执行下一次的循环。continue属于可以正常完成循环的跳转语句。
